Source/Wikitivity.Agent/WikitivityUploadsAgent.cs

```diff
@@ -51,155 +57,130 @@ public override void Execute()
 			{
 				RaiseMessage("Obtaining Case IDs", 1);
 
-				using (IRSAPIClient proxy = Helper.GetServicesManager().CreateProxy<IRSAPIClient>(ExecutionIdentity.System))
+				Task.Run(async () =>
 				{
-					CasesWithWikitivity = wAHelper.GetCasesWithWikitivity(proxy);
-					if (CasesWithWikitivity.Count > 0)
+					CasesWithWikitivity = await wAHelper.GetCasesWithWikitivityOM(proxy);
+				}).Wait();
+				if (CasesWithWikitivity.Count > 0)
+				{
+					RaiseMessage("Found cases with wikitivity installed", 1);
+			
+					foreach (int singleWorkspace in CasesWithWikitivity)
 					{
-						RaiseMessage("Found cases with wikitivity installed", 1);
-						//this means we returned a workspace that has wikitivity installed. I'm of the opinion that we should "cheat" here 
-						//and just pretend like theres only ever going to be one workspace with this installed for now.
+						// Set the workspaceID to this workspace for reuse
+						int workspaceID = singleWorkspace;
+						RaiseMessage("WorkspaceID: " + workspaceID, 1);
 
-						// Form the query to get the specific case we're going to work on
-						Query<RDO> getInitialJobQuery = new Query<RDO>();
-						getInitialJobQuery.ArtifactTypeGuid = WikitivityRDOGUID;
-						getInitialJobQuery.Condition = new TextCondition("Request Url", TextConditionEnum.IsSet);
-						getInitialJobQuery.Fields = FieldValue.AllFields;
-						getInitialJobQuery.ArtifactTypeGuid = WikitivityRDOGUID;
 
+						getInitialJobResultSet = WikitivityAgentHelper.FindAllWikiCases(proxy, workspaceID).Result;
 
-						foreach (int singleWorkspace in CasesWithWikitivity)
-						{
 
+						if (getInitialJobResultSet.TotalCount > 0)
+						{

```

Comment:
This is something called an arrow head.  It starts to make the function hard to test process as you get so deep into the code.
If/ForEach
   If/ForEach
       if/ForEach